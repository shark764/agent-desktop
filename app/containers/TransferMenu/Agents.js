import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import { FormattedMessage } from 'react-intl';
import Icon from 'components/Icon';
import messages from './messages';

function Agents(props) {
  return (
    <React.Fragment>
      {!props.nonVoice &&
          props.showAgentsTransferMenuPreference &&
          props.hasAgentExperienceTransferMenuAgentsViewPermission && (
        <div style={props.styles.transferListDivContainer}>
          <div
            id="agentsExpandCollapseBtn"
            key="agentsListBtn"
            style={[
              { marginTop: '20px' },
              props.styles.expandedTransferHeading,
            ]}
            onClick={() =>
              !props.nonVoice &&
                  props.updateAgentsListVisibleState()
            }
          >
            <FormattedMessage {...messages.agents} />
            {(props.agentsListVisibleState ||
                  props.transferSearchInput.trim() !== '') && (
              <div
                id="refreshAgents"
                key="agentsRefreshBtn"
                style={[
                  props.styles.refresh,
                  props.selectAgents === undefined &&
                        props.styles.refreshInProgress,
                ]}
                onClick={props.refreshAgents}
              >
                    &#8635;
              </div>
            )}
            <Icon
              name="caret"
              style={
                props.agentsListVisibleState
                  ? props.styles.iconOpen
                  : props.styles.iconClosed
              }
            />
          </div>
          {(props.agentsListVisibleState ||
                props.transferSearchInput.trim() !== '') &&
                props.agents}
          <div className="bigSpacer" style={props.styles.lineSpacer} />
        </div>
      )}
    </React.Fragment>
  );
}

Agents.propTypes = {
  nonVoice: PropTypes.bool,
  showAgentsTransferMenuPreference: PropTypes.bool.isRequired,
  hasAgentExperienceTransferMenuAgentsViewPermission: PropTypes.bool.isRequired,
  updateAgentsListVisibleState: PropTypes.func.isRequired,
  agentsListVisibleState: PropTypes.bool,
  transferSearchInput: PropTypes.string,
  selectAgents: PropTypes.array,
  refreshAgents: PropTypes.func.isRequired,
  agents: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  styles: PropTypes.object.isRequired,
};

export default Radium(Agents);
