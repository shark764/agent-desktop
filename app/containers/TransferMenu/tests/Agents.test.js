import React from 'react';
import { shallow } from 'enzyme';
import Agents from '../Agents';

describe('Agents Component render', () => {
  const agentsList = [];
  agentsList.push(
    <div
      key='1'
      id='1'
      className="readyAgentTransferListItem"
      style={{}}
      title="agentName"
    >
      <div style={{}} />
      <span style={{}}>
agentName
      </span>
    </div>
  );

  it('renders Agents component with permissions, showing menu agent transfer, without searching input', () => {
    const component = shallow(
      <Agents
        showAgentsTransferMenuPreference
        hasAgentExperienceTransferMenuAgentsViewPermission
        updateAgentsListVisibleState={() => 'mock updateAgentsListVisibleState'}
        agentsListVisibleState
        transferSearchInput="agentName"
        selectAgents={undefined}
        refreshAgents={() => 'mock refreshAgents'}
        agents={agentsList}
        styles={{}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders Agents component without permissions, it should not render anything', () => {
    const component = shallow(
      <Agents
        showAgentsTransferMenuPreference
        hasAgentExperienceTransferMenuAgentsViewPermission={false}
        updateAgentsListVisibleState={() => 'mock updateAgentsListVisibleState'}
        agentsListVisibleState
        transferSearchInput="agentName"
        selectAgents={undefined}
        refreshAgents={() => 'mock refreshAgents'}
        agents={agentsList}
        styles={{}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders Agents component with permissions, but not showing the menu preference, it should not render anything', () => {
    const component = shallow(
      <Agents
        showAgentsTransferMenuPreference={false}
        hasAgentExperienceTransferMenuAgentsViewPermission
        updateAgentsListVisibleState={() => 'mock updateAgentsListVisibleState'}
        agentsListVisibleState
        transferSearchInput="agentName"
        selectAgents={undefined}
        refreshAgents={() => 'mock refreshAgents'}
        agents={agentsList}
        styles={{}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders Agents component with permissions, not showing the refresh buttons', () => {
    const component = shallow(
      <Agents
        showAgentsTransferMenuPreference
        hasAgentExperienceTransferMenuAgentsViewPermission
        updateAgentsListVisibleState={() => 'mock updateAgentsListVisibleState'}
        agentsListVisibleState={false}
        transferSearchInput=""
        selectAgents={undefined}
        refreshAgents={() => 'mock refreshAgents'}
        agents={agentsList}
        styles={{}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders Agents component with permissions, with an empty agentsList', () => {
    const component = shallow(
      <Agents
        showAgentsTransferMenuPreference
        hasAgentExperienceTransferMenuAgentsViewPermission
        updateAgentsListVisibleState={() => 'mock updateAgentsListVisibleState'}
        agentsListVisibleState
        transferSearchInput="agentName"
        selectAgents={undefined}
        refreshAgents={() => 'mock refreshAgents'}
        agents={[]}
        styles={{}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
