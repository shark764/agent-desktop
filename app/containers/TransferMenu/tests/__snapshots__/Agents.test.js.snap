// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agents Component render renders Agents component with permissions, but not showing the menu preference, it should not render anything 1`] = `<Fragment />`;

exports[`Agents Component render renders Agents component with permissions, not showing the refresh buttons 1`] = `
<Fragment>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      id="agentsExpandCollapseBtn"
      key=".$agentsListBtn"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <FormattedMessage
        defaultMessage="Agents"
        id="app.containers.TransferMenu.agents"
        key=".0"
        values={Object {}}
      />
      <Icon
        key=".2"
        name="caret"
      />
    </div>
    <div
      className="bigSpacer"
      key=".2"
    />
  </div>
</Fragment>
`;

exports[`Agents Component render renders Agents component with permissions, showing menu agent transfer, without searching input 1`] = `
<Fragment>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      id="agentsExpandCollapseBtn"
      key=".$agentsListBtn"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <FormattedMessage
        defaultMessage="Agents"
        id="app.containers.TransferMenu.agents"
        key=".0"
        values={Object {}}
      />
      <div
        data-radium={true}
        id="refreshAgents"
        key=".$agentsRefreshBtn"
        onClick={[Function]}
        style={Object {}}
      >
        ↻
      </div>
      <Icon
        key=".2"
        name="caret"
      />
    </div>
    <div
      className="readyAgentTransferListItem"
      data-radium={true}
      id="1"
      key=".1:$1"
      style={Object {}}
      title="agentName"
    >
      <div
        data-radium={true}
        key=".0"
        style={Object {}}
      />
      <span
        data-radium={true}
        key=".1"
        style={Object {}}
      >
        agentName
      </span>
    </div>
    <div
      className="bigSpacer"
      key=".2"
    />
  </div>
</Fragment>
`;

exports[`Agents Component render renders Agents component with permissions, with an empty agentsList 1`] = `
<Fragment>
  <div
    data-radium={true}
  >
    <div
      data-radium={true}
      id="agentsExpandCollapseBtn"
      key=".$agentsListBtn"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      <FormattedMessage
        defaultMessage="Agents"
        id="app.containers.TransferMenu.agents"
        key=".0"
        values={Object {}}
      />
      <div
        data-radium={true}
        id="refreshAgents"
        key=".$agentsRefreshBtn"
        onClick={[Function]}
        style={Object {}}
      >
        ↻
      </div>
      <Icon
        key=".2"
        name="caret"
      />
    </div>
    <div
      className="bigSpacer"
      key=".2"
    />
  </div>
</Fragment>
`;

exports[`Agents Component render renders Agents component without permissions, it should not render anything 1`] = `<Fragment />`;
